aerogear/unifiedpush-operator
# INTLY-3288 - UPS CSV

## Motivation
To allow Integreatly to consume the Unified Push Server Operator, a Cluster Service Version (CSV) would need to be generated. This pull request adds the generated csv that is to be used by integreatly.

Jira tickets include:
* https://issues.jboss.org/browse/INTLY-3288
* https://issues.jboss.org/browse/INTLY-3691

## What
Adds the Cluster Service Version (CSV) for the Unified Push Server Operator

## Why
This was added so the UPS operator can be used as part of the Integreatly Operator install

## How
This was done via the `operator-sdk olm-catalog gen-csv --csv-version=0.2.0` command for generating the base csv.

The csv was then further edited to include information on customresourcedefinitions, description, version, provider etc and the channel was updated to the `integreatly` channel to match the other csv in the [integreatly manifest repo](https://github.com/integr8ly/manifests).

The operator container image used in this CSV points to an image containing the `oauth-proxy` update - https://github.com/aerogear/unifiedpush-operator/pull/75 

## Verification Steps
* On a Openshift 4 cluster, create a new project 
* Create a operator source that points to my app registry:
```
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: integreatly-operators
  namespace: openshift-marketplace
spec:
  authorizationToken: {}
  displayName: Integreatly Operators
  endpoint: 'https://quay.io/cnr'
  publisher: Integreatly Publisher
  registryNamespace: kevfan
  type: appregistry
```
* Browse to the operator hub and verifiy that a Unifiedpush Operator with version `0.2.0` is present
* Verify that the `unifiedpush-operator` pod is created
* Kickstart the installation using the following `UnifiedPushServer.cr` as an example:
```
apiVersion: push.aerogear.org/v1alpha1
kind: UnifiedPushServer
metadata:
  name: example-unifiedpushserver
spec:
  useMessageBroker: false
```
* Verify the resources are created successully
* Visit the exposed ups route to ensure it works

**Note: As this is a copy of the csv that are going to be added to integreatly manifest repo, this should be merged when only the following pull requests are merged:**
* https://github.com/integr8ly/manifests/pull/70
* https://github.com/integr8ly/integreatly-operator/pull/127
## Checklist:


## Progress
